main.ml

```diff
@@ -361,6 +361,17 @@ let process root_dir name =
         R.Enum (List.map ~f:(compat_out fields) l)
     | _ -> rpc
   in
+  let compat_uri device_config ~f =
+    if !version = Some pvs_version then
+      match List.Assoc.find ~equal:String.equal device_config "uri" with
+      | None ->
+        return (Error (missing_uri ()))
+      | Some uri ->
+        return (Ok (device_config |> f |> compat_out ["uri", R.String uri]))
```

Comment:
So does this mean that for the 3.0 version there might be additional things in the dict, but that will be ignored by those scripts?

Reply:
Yeah the `uri` was mandatory for 3.0, and you could have extra (backend-specific) info in the dict. Running the functional tests on pvsproxy passed, so this should be fine.
What I changed here is that `uri` is no longer mandatory in `device-config`.

Reply:
So in general, if we pass to the script a dict with some extra parameters it doesn't know about, that won't cause an error, they will be ignored (for example device_config in this case)?

Reply:
Actually this might need more testing, will schedule some pvs functional tests on the probe branch again...

Reply:
Tests looked good, so I think we can safely assume that unknown fields in the dict are ignored.